File: Proxy/Java/ProtectionProxy/src/DocumentProxy.java
public class DocumentProxy implements IDocument{
    private SecretDocument secretDocument;
    private String password;

    public DocumentProxy(String password) {
        this.password = password;
    }

    @Override
    public void display() {
        if ("1234".equals(this.password)) {
            if (secretDocument == null) {
                secretDocument = new SecretDocument();
            }
            secretDocument.display();
        } else {
            System.out.println("Access denied: Incorrect password.");
        }
    }
}

File: Proxy/ProtectionProxy/ProtectionProxy/DocumentProxy.cs
namespace ProtectionProxy
{
    public class DocumentProxy : IDocument
    {
        private SecretDocument _secretDocument;
        private string _password;

        public DocumentProxy(string password)
        {
            _password = password;
        }

        public void Display()
        {
            if (_password == "1234")
            {
                if (_secretDocument == null)
                {
                    _secretDocument = new SecretDocument();
                }
                _secretDocument.Display();
            }
            else
            {
                Console.WriteLine("Access denied: Incorrect password.");
            }
        }
    }
}
